### PR TITLE
Added methods to manage async operation

### PR DESCRIPTION
**Pull request by yogevmalcaym**
Added methods to control async operations. Also Edited the example.js file to cover the async method.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,40 +1,100 @@
...
+        items,
+        destination,
+        sourceDirPath
+      });
+    });
+
+    const allPromises = promisesArrays.reduce((acc, arr) => [...acc, ...arr]);
+
+    await Promise.all(allPromises);
+    return "Done copyFiles successfully";
```
Comment (GoorLavi) [inline index.js]: @yogevmalcaym 
Please do not return meaningless strings 
in case of a crash or having an error he will know it

File: index.js [hunk 2]
```diff
@@ -1,40 +1,100 @@
...
+  } catch (error) {
+    console.error(error);
+    return error;
+  }
 };
 
-module.exports = {
-    copyFilesSync,
-};
+const copyItemsAsync = ({
```
Comment (GoorLavi) [inline index.js]: Please change `copyItemsAsync` -> `copyItems`

File: index.js [hunk 3]
```diff
@@ -1,40 +1,100 @@
...
+      const items = [...foldersAndFiles, ...getAllFilteredItems(sourceDirPath, allDirectories, allFiles)];
 
+      return copyItemsAsync({
+        items,
+        destination,
+        sourceDirPath
+      });
+    });
+
+    const allPromises = promisesArrays.reduce((acc, arr) => [...acc, ...arr]);
```
Comment (GoorLavi) [inline index.js]: if I'm not wrong you can use `flat()`

File: index.js [hunk 4]
```diff
@@ -1,40 +1,100 @@
 const fs = require('fs-extra');
 
-
 const getAllFilteredItems = (path, isAllDirectory, isAllFiles) => {
 
-    return (fs.readdirSync(path) || []).filter(item => {
-        const isDirectory = fs.statSync(`${path}/${item}`).isDirectory();
```
Comment (GoorLavi) [inline index.js]: can you please change your indent to 4 spaces?

File: index.js [hunk 5]
```diff
@@ -1,40 +1,100 @@
...
+        resolve();
+      });
+    })
+  );
+
+
+const copyFiles = async filesData => {
 
+  try {
+    const promisesArrays = Object.keys(filesData).map(sourceDirPath => {
```
Comment (GoorLavi) [inline index.js]: can be better if you will change it to `for` loop.
this way you could save iteration.

Comment (GoorLavi) [reply index.js]: please do it in `copyFilesSync` too

**Review by GoorLavi — COMMENTED** [review]
Greate job bro

File: index.js (hunk 4 above)
Comment (yogevmalcaym) [reply index.js]: You mean tab instead of line space?

Comment (GoorLavi) [reply index.js]: @yogevmalcaym no there are indent options use 4 spaces

File: index.js [hunk 6]
```diff
@@ -1,40 +1,100 @@
...
+      const {
+        options: {
+          allFiles,
+          allDirectories
+        } = {},
+        destination,
+        foldersAndFiles = []
+      } = folderObject;
 
-                const items = [...foldersAndFiles, ...getAllFilteredItems(sourceDirPath, allDirectories, allFiles)];
```
Comment (GoorLavi) [inline index.js]: i think `Array.concat` will be better.

**Review by GoorLavi — COMMENTED** [review]
a

**Pull request merged**